File: CSharp/HtmlDocCleaner/DocCleanerModifier.cs
using HtmlWalker;
using HtmlWalker.Tag.Html;
using HtmlWalker.Tag.Util;
using HtmlWalker.Visitor;
using static HtmlWalker.Platform.WalkerPlatform;

namespace HtmlDocCleaner
{
    public class DocCleanerModifier : EchoModifer
    {
        private WalkerFactory factory;
        public DocCleanerModifier(WalkerFactory factory)
        {
            this.factory = factory;
        }

        public override ITag Visit(Comment adapter) => new Skip(factory, adapter);

        public override ITag Visit(TagHead adapter) => new Skip(factory, adapter) { SkipTree = true };

        public override ITag Visit(TagDiv adapter) => new Skip(factory, adapter);
    }
}

File: Java/htmlwalker/src/main/java/htmldoccleaner/DocCleanerModifier.java
package htmldoccleaner;

import htmlwalker.ITag;
import htmlwalker.platform.WalkerFactory;
import htmlwalker.tag.html.*;
import htmlwalker.tag.util.*;
import htmlwalker.visitor.EchoModifier;

public class DocCleanerModifier extends EchoModifier
{
    private WalkerFactory factory;

    public DocCleanerModifier(WalkerFactory factory)
    {
        this.factory = factory;
    }

    @Override
    public ITag visit(Comment adapter) { return new Skip(factory, adapter); }

    @Override
    public ITag visit(TagHead adapter)
    {
        var skip = new Skip(factory, adapter);
        skip.skipTree = true;
        return skip;
    }

    @Override
    public ITag visit(TagDiv adapter) { return new Skip(factory, adapter); }
}
